tests/filecheck/dialects/symref/ops.mlir

```diff
@@ -0,0 +1,71 @@
+// RUN: XDSL_ROUNDTRIP
+
+builtin.module {
+  func.func @symref_test() {
+    // Declare some symbols
+    "symref.declare"() {"sym_name" = "counter"} : () -> ()
+    "symref.declare"() {"sym_name" = "flag"} : () -> ()
+
+    // Test basic fetch operations
+    %0 = "symref.fetch"() {"symbol" = @counter} : () -> i32
+    %1 = "symref.fetch"() {"symbol" = @flag} : () -> i1
+
+    // Create some values to update with
+    %c42 = arith.constant 42 : i32
+    %true = arith.constant true
+
+    // Test update operations
+    "symref.update"(%c42) {"symbol" = @counter} : (i32) -> ()
+    "symref.update"(%true) {"symbol" = @flag} : (i1) -> ()
+
+    // Test fetch after update
+    %3 = "symref.fetch"() {"symbol" = @counter} : () -> i32
+    %4 = "symref.fetch"() {"symbol" = @flag} : () -> i1
+
+    func.return
+  }
+
+  func.func @symref_complex_types() {
+    // Test with more complex types
+    "symref.declare"() {"sym_name" = "tensor_ref"} : () -> ()
+    "symref.declare"() {"sym_name" = "memref_ref"} : () -> ()
+
+    %tensor = "test.op"() : () -> tensor<4x4xf32>
+    %memref = "test.op"() : () -> memref<8xi64>
+
+    "symref.update"(%tensor) {"symbol" = @tensor_ref} : (tensor<4x4xf32>) -> ()
+    "symref.update"(%memref) {"symbol" = @memref_ref} : (memref<8xi64>) -> ()
+
+    %7 = "symref.fetch"() {"symbol" = @tensor_ref} : () -> tensor<4x4xf32>
+    %8 = "symref.fetch"() {"symbol" = @memref_ref} : () -> memref<8xi64>
+
+    func.return
+  }
+}
+
+// CHECK: builtin.module {
+// CHECK-NEXT:   func.func @symref_test() {
```

Comment:
I'm more of a believer in interleaving the check statements around the code. I think it makes diffs far easier in the future.